Tests/kaas/k8s-node-distribution/k8s-node-distribution-check.py

```diff
@@ -0,0 +1,258 @@
+
+    nodes = list()
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        core_api = kubernetes_asyncio.client.CoreV1Api(api)
+        result = await core_api.list_node()
+        for item in result.items:
+            node = dict()
+            node_labels = item.metadata.labels
+            for label in labels:
+                try:
+                    node[label] = node_labels[label]
+                except KeyError:
+                    # Node is not labelled as a control plane node
+                    pass
+            nodes.append(node)
+
+    return nodes
+
+
+def compare_labels(node_list, labels, node_type="master"):
+
+    label_data = dict()
+
+    for node in node_list:
+        for key in labels:
+            if key not in label_data:
+                label_data[key] = set()
+            try:
+                label_data[key].add(node[key])
+            except KeyError:
+                logger.warning(f"The label for {labels[key].split('/')[1]}s don't seem to be set for all nodes.")
+
+    for label in labels:
+        if len(labels[label]) <= 1:
+            if len(labels[label]) == 1:
+                logger.warning(f"There seems to be no distribution across multiple {label.split('/')[1]}s "
+                               "or labels aren't set correctly across nodes.")
+            elif len(labels[label]) == 0:
+                logger.warning(f"The label for {label.split('/')[1]}s don't seem to be set for all nodes.")
+        else:
+            logger.info(f"The nodes are distributed across {str(len(labels[label]))} {label.split('/')[1]}s.")
+            return
+
+    if node_type == "master":
+        raise DistributionException("The distribution of nodes described in the standard couldn't be detected.")
+    elif node_type == "worker":
+        logger.warning("No node distribution could be detected for the worker nodes. "
+                       "This produces only a warning, since it is just a recommendation.")
+        return
+
+
+async def main(argv):
+    try:
+        config = initialize_config(parse_arguments(argv))
+    except (OSError, ConfigException, HelpException) as e:
+        if hasattr(e, 'message'):
+            logger.error(e.message)
+        print_usage()
+        return 1
+
+    # It is important to note, that the order of these labels matters for this test.
+    # Since we want to check if nodes are distributed, we want to do this from bigger
+    # infrastructure parts to smaller ones. So we first look if nodes are distributed
+    # across regions, then zones and then hosts. If one of these requirements is fulfilled,
+    # we probably don't need to check anymore, since a distribution was already detected.
+    labels = (
+        "topology.kubernetes.io/region",
+        "topology.kubernetes.io/zone",
+        "kubernetes.io/hostname",
+    )
+
+    nodes = await get_k8s_cluster_labelled_nodes(config.kubeconfig, labels + tuple("node-role.kubernetes.io/control-plane"))
```

Comment:
```suggestion
    nodes = await get_k8s_cluster_labelled_nodes(config.kubeconfig, labels + ("node-role.kubernetes.io/control-plane", ))
```

Your version does strange things:

```console
$ ipython3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.31.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: tuple("node-role.kubernetes.io/control-plane")
Out[1]: 
('n',
 'o',
 'd',
 'e',
 '-',
 'r',
 'o',
 'l',
 'e',
 '.',
 'k',
 'u',
 'b',
 'e',
 'r',
 'n',
 'e',
 't',
 'e',
 's',
 '.',
 'i',
 'o',
 '/',
 'c',
 'o',
 'n',
 't',
 'r',
 'o',
 'l',
 '-',
 'p',
 'l',
 'a',
 'n',
 'e')
```

Reply:
Interesting, thank you.